src/app/pages/submission-pages/data-page/data-page.component.ts

```diff
@@ -44,14 +44,15 @@ export class DataPageComponent implements OnInit {
     this.activeSubmission = this.submissionsService.getActiveSubmission();
     this.token = this.tokenService.getToken();
 
-    this.submissionsService.getActiveSubmissionsFiles().subscribe(data => {
-      this.files = data;
-      // format the file status string and store it as status_label for display
-      this.files['_embedded']['files'].forEach(element => {
-        element.status_label = this.formatFileStatus(element.status);
-      });
-    });
-
+    this.fileService.getActiveSubmissionsFiles(this.activeSubmission).subscribe(
+      (data) => {
+        this.files = data;
+        // format the file status string and store it as status_label for display
+        this.files['_embedded']['files'].forEach(element => {
```

Comment:
I do suggest that we remove these 3 lines and from the template we do something like 

{{formatFileStatus(file.status)}}

Reply:
OK, I will do that, but I have not changed these lines.
I just moved the service method to the relevant service and use it from there.

Reply:
I fixed it

Reply:
Yea I see, I just suggested a better implementation for this.